src/client.ts

```diff
@@ -4627,22 +4627,24 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             threadId = null;
         }
         const reason = opts?.reason;
+        let withRelTypes = opts?.with_rel_types ?? opts?.with_relations;
 
-        if (
-            opts?.with_relations &&
-            this.canSupport.get(Feature.RelationBasedRedactions) === ServerSupport.Unsupported
-        ) {
+        if (typeof withRelTypes === "string") {
+            withRelTypes = [withRelTypes];
+        }
+
+        if (withRelTypes && this.canSupport.get(Feature.RelationBasedRedactions) === ServerSupport.Unsupported) {
             throw new Error(
                 "Server does not support relation based redactions " +
                     `roomId ${roomId} eventId ${eventId} txnId: ${txnId} threadId ${threadId}`,
             );
         }
 
-        const withRelations = opts?.with_relations
+        const withRelTypesContent = withRelTypes
             ? {
                   [this.canSupport.get(Feature.RelationBasedRedactions) === ServerSupport.Stable
                       ? MSC3912_RELATION_BASED_REDACTIONS_PROP.stable!
-                      : MSC3912_RELATION_BASED_REDACTIONS_PROP.unstable!]: opts?.with_relations,
+                      : MSC3912_RELATION_BASED_REDACTIONS_PROP.unstable!]: withRelTypes,
               }
             : {};
```

Comment:
how about enclosing all this in an `if (!!withRelTypes)` block, to save repeatedly checking it?